Mathlib/Tactic/TFAE.lean

```diff
@@ -24,39 +24,38 @@ syntax impArrow := " → " <|> " ↔ " <|> " ← "
 
 /--
 `tfae_have` introduces hypotheses for proving goals of the form `TFAE [P₁, P₂, ...]`. Specifically,
-`tfae_have i arrow j` introduces a hypothesis of type `Pᵢ arrow Pⱼ` to the local context,
-where `arrow` can be `→`, `←`, or `↔`. Note that `i` and `j` are natural number indices (beginning
-at 1) used to specify the propositions `P₁, P₂, ...` that appear in the `TFAE` goal list. A proof
-is required afterward, typically via a tactic block.
+`tfae_have i <arrow> j` introduces a hypothesis of type `Pᵢ <arrow> Pⱼ` to the local context,
+where `<arrow>` can be `→`, `←`, or `↔`. Note that `i` and `j` are natural number indices (beginning
+at 1) used to specify the propositions `P₁, P₂, ...` that appear in the `TFAE` goal list.
+
+A proof can be supplied immediately via `:=` or afterwards via a tactic block, like mathlib's
+`have`.
 
 ```lean
 example (h : P → R) : TFAE [P, Q, R] := by
-  tfae_have 1 → 3
-  · exact h
+  tfae_have 1 → 3 := h
   ...
 ```
 The resulting context now includes `tfae_1_to_3 : P → R`.
 
 The introduced hypothesis can be given a custom name, in analogy to `have` syntax:
 ```lean
-tfae_have h : 2 ↔ 3
+tfae_have h' : 2 ↔ 3
 ```
 
 Once sufficient hypotheses have been introduced by `tfae_have`, `tfae_finish` can be used to close
-the goal.
+the goal. The following example uses both tactic block and `:=` syntax for `tfae_have`.
 
 ```lean
 example : TFAE [P, Q, R] := by
-  tfae_have 1 → 2
-  · /- proof of P → Q -/
+  tfae_have 1 → 2 := sorry /- term of type P → Q -/
   tfae_have 2 → 1
-  · /- proof of Q → P -/
-  tfae_have 2 ↔ 3
-  · /- proof of Q ↔ R -/
+  · admit /- proof of Q → P -/
+  tfae_have 2 ↔ 3 := by admit /- proof of Q ↔ R -/
   tfae_finish
 ```
 -/
-syntax (name := tfaeHave) "tfae_have " (ident " : ")? num impArrow num : tactic
+syntax (name := tfaeHave) "tfae_have " (ident " : ")? num impArrow num (":=" term)? : tactic
```

Comment:
Can you recycle the `declVal` syntax here?

Reply:
I think `declVal` is for fully generic declarations, and allows some syntax we don't want (like`where` and termination-proving syntax).

Though, we might want to recreate the rest of `letDecl` to allow patterns and match alts! That could be helpful with proving iffs. (However, we can't literally recycle the code, since we demand `num impArrow num` in the middle of everything here...) But it's slightly more involved (I'd need to familiarize myself with how we elaborate `let`s in general), so I'd rather save it for another PR.

Reply:
Thanks for the explanation: it cleared up some doubts that I did not I had!